vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,321 @@
+package edu.asu.diging.vspace.web.staff;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.asu.diging.vspace.core.data.ModuleRepository;
+import edu.asu.diging.vspace.core.data.SpaceRepository;
+import edu.asu.diging.vspace.core.data.display.ModuleLinkDisplayRepository;
+import edu.asu.diging.vspace.core.data.display.SpaceLinkDisplayRepository;
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.model.display.impl.ModuleLinkDisplay;
+import edu.asu.diging.vspace.core.model.display.impl.SpaceLinkDisplay;
+import edu.asu.diging.vspace.core.model.impl.Module;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.web.api.ImageApiController;
+
+/**
+ * 
+ * @author Avirup Biswas
+ *
+ */
+@Controller
+public class SpaceOverviewController {
+    @Autowired
+    private SpaceRepository spaceRepo;
+
+    @Autowired
+    private ModuleRepository moduleRepo;
+
+    @Autowired
+    private SpaceLinkDisplayRepository spaceLinkDisplayRepository;
+
+    @Autowired
+    private ModuleLinkDisplayRepository moduleLinkDisplayRepository;
+
+    @RequestMapping("/staff/overview")
+    public String spaceOverview(HttpServletRequest request, Model model) throws JsonProcessingException {
```

Comment:
this is all still very hard to follow, especially since you're mixing business logic (get a graph of all spaces and modules) and presentation (turn that graph into json). The business logic should actually go into a manager/service class (I would create a new one for that). This will also move the repositories in the business layer logic and it will make it easy to reuse for other purposes if needed. 
Your controller should then just turn the result into json. I would also recommend using ids for the edges as it should make the code that generates the json a lot easier (if you keep using the force layout, then this should help with that: http://bl.ocks.org/nitaku/7483341).

Reply:
I have to create link node json where I have to specify the source and target of each node .I don't have to write any extra code for edge creation. The link you have provided are using extra code to create link using id. We can discuss regarding this.
Note:I can see for each node in the node json there is a unique index starting from 0 gets appended to each json entry of the node json array.